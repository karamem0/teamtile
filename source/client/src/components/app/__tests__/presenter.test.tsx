//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

jest.mock('@azure/msal-browser', () => ({
  PublicClientApplication: jest.fn()
}));

jest.mock('@azure/msal-react', () => ({
  MsalProvider: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="MsalProvider">
      {children}
    </div>
  )
}));

jest.mock('../../app-insights', () => ({
  AppInsights: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="AppInsights">
      {children}
    </div>
  )
}));

jest.mock('../../router', () => ({
  Router: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="Router">
      {children}
    </div>
  )
}));

jest.mock('../../../config/auth-config', () => ({
  authConfig: {}
}));

jest.mock('../../../contexts/error-context', () => ({
  ErrorContextProvider: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="ErrorContextProvider">
      {children}
    </div>
  )
}));

jest.mock('../../theme-provider', () => ({
  ThemeProvider: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="ThemeProvider">
      {children}
    </div>
  )
}));

import React from 'react';

import { render, screen } from '@testing-library/react';

import App from '../presenter';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('App', () => {

  it('create shapshot', () => {
    render(<App />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
