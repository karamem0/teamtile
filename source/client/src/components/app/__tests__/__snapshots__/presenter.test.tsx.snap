// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App create shapshot 1`] = `
<body>
  <div>
    <div
      data-testid="AppInsights"
    >
      <div
        data-testid="MsalProvider"
      >
        <div
          data-testid="ThemeProvider"
        >
          <div
            data-testid="ErrorContextProvider"
          >
            <div
              data-testid="Router"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
